src/functions-and-arrays.js

```diff
@@ -1,19 +1,96 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2 || num1 === num2) {
+    return num1;
+  } 
+  return num2;
+};
+
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(words) {
+  if (words.length === 0) {
+    return null;
+  }
+  let longestWord = '';
+  for (word of words) {
+    if (word.length > longestWord.length) {
+      longestWord = word;
+    } 
+  }
+  return longestWord;
+}
+
+console.log(findLongestWord(words));
 // Iteration #3: Calculate the sum
 
-const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+const numbers = [6, 12, 1, 18, 13, 16, 2,[2,3,5], 1, 8, 10];
+
+function sumNumbers(numbers) {
+  let sum = 0;
+  for (number of numbers) {
+    sum += number;
+  }
+  return sum;
+}
+
+let arr = [3,4,2]
+
+
+function sum(arr){
+  let sum =0;
+  for (item of arr) {
+    if (typeof item === 'string') {
+      sum += item.length;
+    } else if (typeof item === 'object') {
+        throw Error("Unsupported data type sir or ma'am") //"Error: Oh no! An array/object."; 
+    } else {
+      sum += item;
+    }
+  }
+  return sum;
+}
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10]
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbers) {
+  if (numbers.length === 0){
+    return null;
+  }
+  sum = sumNumbers(numbers);
```

Comment:
Use a variable keyword (let/const/var) for declaring variables. In this case I would use const, as "sum" is not reassigned.